packages/portfolio-planner/src/engine.ts

```diff
@@ -46,6 +65,72 @@ const stripPrefix = (prefix: string, str: string) => {
   return str.slice(prefix.length);
 };
 
+// Track active CCTP watchers to avoid duplicates
+type CCTPWatcherKey = `${string}-${string}`; // portfolioKey-chain
+const activeCCTPWatchers = new Set<CCTPWatcherKey>();
+
+const launchCCTPWatchers = async (
+  portfolioKey: string,
+  pendingTransfers: { [chain: string]: CCTPTransfer },
+) => {
+  for (const [chainName, transfer] of Object.entries(pendingTransfers)) {
+    const evmChain = chainName as EVMChain;
+    if (!evmChain) {
+      console.warn(
+        `[CCTP] Unsupported chain for CCTP monitoring: ${chainName}`,
+      );
+      continue;
+    }
+
+    // TODO: What if there are more than 1 pending CCTP transfers for a portfolio
+    const watcherKey: CCTPWatcherKey = `${portfolioKey}-${chainName}`;
+    if (activeCCTPWatchers.has(watcherKey)) {
+      console.log(
+        `[CCTP] Watcher already active for portfolio ${portfolioKey} on ${chainName}`,
+      );
+      continue;
+    }
+
+    console.log(
+      `[CCTP] Starting watcher for portfolio ${portfolioKey} on ${chainName}`,
+      {
+        receiver: transfer.receiver,
+        amount: transfer.amount,
+        caip: transfer.caip,
+      },
+    );
+
+    activeCCTPWatchers.add(watcherKey);
+
+    watchCCTPMint({
+      chain: evmChain,
+      recipient: transfer.receiver,
+      expectedAmount: BigInt(transfer.amount),
+    })
```

Comment:
how does `watchCCTPMint` access any EVM chain state? I don't see any I/O capabilities passed in.